src/rust/engine/process_execution/src/remote.rs

```diff
@@ -733,6 +748,28 @@ impl CommandRunner {
   }
 }
 
+fn maybe_add_workunit(result_cached: bool, name: &str, start_time: &Timespec, end_time: &Timespec, parent_id: Option<String>, workunit_store: &WorkUnitStore) {
+//  TODO: workunits for scheduling, fetching, executing and uploading should be recorded
+//   only if '--reporting-zipkin-trace-v2' is set
+  if !result_cached {
+    let workunit = WorkUnit {
+      name: String::from(name),
```

Comment:
We probably want to use slightly more verbose names, something like "remote execution worker input fetching" and similar